langchain/callbacks/arize_callback.py

```diff
@@ -1,217 +1,314 @@
-
-                # Define the columns and data
-                columns = [
-                    "prediction_ts",
-                    "response",
-                    "prompt",
-                    "response_vector",
-                    "prompt_vector",
-                    "prompt_token",
-                    "completion_token",
-                    "total_token",
-                ]
-                data = [
-                    [
-                        pred_timestamp,
-                        response_text,
-                        prompt,
-                        response_embedding[0],
-                        prompt_embedding[0],
-                        self.prompt_tokens,
-                        self.total_tokens,
-                        self.completion_tokens,
-                    ]
-                ]
-
-                # Create the DataFrame
-                df = pd.DataFrame(data, columns=columns)
-
-                # Declare prompt and response columns
-                prompt_columns = EmbeddingColumnNames(
-                    vector_column_name="prompt_vector", data_column_name="prompt"
-                )
-
-                response_columns = EmbeddingColumnNames(
-                    vector_column_name="response_vector", data_column_name="response"
-                )
-
-                schema = Schema(
-                    timestamp_column_name="prediction_ts",
-                    tag_column_names=[
-                        "prompt_token",
-                        "completion_token",
-                        "total_token",
-                    ],
-                    prompt_column_names=prompt_columns,
-                    response_column_names=response_columns,
-                )
-
-                response_from_arize = self.arize_client.log(
-                    dataframe=df,
-                    schema=schema,
-                    model_id=self.model_id,
-                    model_version=self.model_version,
-                    model_type=ModelTypes.GENERATIVE_LLM,
-                    environment=Environments.PRODUCTION,
-                )
-                if response_from_arize.status_code == 200:
-                    print("✅ Successfully logged data to Arize!")
-                else:
-                    print(f'❌ Logging failed "{response_from_arize.text}"')
-
-    def on_llm_error(
-        self, error: Union[Exception, KeyboardInterrupt], **kwargs: Any
-    ) -> None:
-        """Do nothing."""
-        pass
+
+def _import_package(package_name: str) -> ModuleType:
+    try:
+        package = importlib.import_module(package_name)
+    except ImportError:
+        raise ImportError(f"The {package_name} package must be installed.")
+    return package
+
+
+def validate_package_version(
+    package: ModuleType,
+    min_supported_version: Optional[str],
+    max_supported_version: Optional[str],
+) -> None:
+    # TODO: implement package version validation at runtime
+    ...
```

Comment:
I think we should aim to be 100% backwards compatible for the SDK and only add this if truly needed.